### PR TITLE
tidb_query: document coprocessor codegen

### PR DESCRIPTION
**Pull request by nrc**
###  What have you changed?

Added some docs, tidied up code in very minor ways

PTAL @breeswish @dcalvin 

###  What is the type of the changes?

- Engineering (engineering change which doesn't change any feature or fix any issue)
- Misc (other changes)

###  How is the PR tested?

doctests

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

no

###  Does this PR affect `tidb-ansible`?

no

### CODE REVIEW

File: components/tidb_query/src/aggr_fn/impl_bit_op.rs [hunk 1]
```diff
@@ -11,6 +11,7 @@ use crate::codec::mysql::Tz;
 use crate::expr::EvalContext;
 use crate::rpn_expr::{RpnExpression, RpnExpressionBuilder};
 use crate::Result;
+use tikv_util::box_try;
```
Comment (breeswish) [inline components/tidb_query/src/aggr_fn/impl_bit_op.rs]: I would rather prefer not to import it.. It is used so frequently that explicit import only introduce inconvenience.

Comment (nrc) [reply components/tidb_query/src/aggr_fn/impl_bit_op.rs]: Addressed.

**Comment by nrc** [comment]
@breeswish comment addressed, PTAL

**Comment by siddontang** [comment]
thanks @nrc 

please assign some labels to your PR

File: components/tidb_query_codegen/src/rpn_function.rs [hunk 2]
```diff
@@ -1,11 +1,125 @@
 // Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
 
+//! Implementation of the `rpn_fn` attribute macro.
+//!
+//! The macro should be applied to a function:
+//!
+//! ```rust
+//! #[rpb_fn]
```
Comment (breeswish) [inline components/tidb_query_codegen/src/rpn_function.rs]: typo: it should be `rpn_fn`

File: components/tidb_query_codegen/src/rpn_function.rs [hunk 3]
```diff
@@ -1,11 +1,125 @@
...
+//!
+//! * `ctx: &mut expr::EvalContext`
+//! * `output_rows: usize`
+//! * `args: &[rpn_expr::RpnStackNode<'_>]`
+//! * `extra: &mut rpn_expr::RpnFnCallExtra<'_>`
+//!
+//! E.g., `#[rpn_fn(capture = [ctx, extra])]`.
+//!
+//!
+//! ## Generated code
```
Comment (breeswish) [inline components/tidb_query_codegen/src/rpn_function.rs]: Actually there are documentations in `components/tidb_query/src/rpn_expr/types`. Would you like to make some integrations like remove some duplicate contents or give some links?

Comment (nrc) [reply components/tidb_query_codegen/src/rpn_function.rs]: I plan to cover the other coprocessor crates in follow-up PRs and will include some links, etc. then. Is that OK?

Comment (breeswish) [reply components/tidb_query_codegen/src/rpn_function.rs]: @nrc I mean, comments in `components/tidb_query/src/rpn_expr/types` already partly covers how `rpn_fn` works. Maybe you can consider consolidate them together?

**Comment by nrc** [comment]
@breeswish I have combined the documentation from the two locations, PTAL.

**Pull request merged**